walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -28,44 +42,83 @@
   private static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
   private static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
 
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+  public Collection<TimeRange> query(
+      Collection<Event> events, MeetingRequest request) {
+
+    // Array of queries representing times to meet.
+    List<TimeRange> optionalTimeRanges;
 
     // Array of queries representing times to meet.
-    List<TimeRange> query = new ArrayList<>();
+    List<TimeRange> mandatoryTimeRanges;
 
     int desiredDuration = (int) request.getDuration();
 
-    // Collection of string representing who is to attend the requested meeting.
-    Collection<String> requestAttendees = request.getAttendees();
+    // Array of queries representing times to meet.
+    List<TimeRange> allTimeRanges;
 
     // Check if duration of event is longer than a whole day and if so return empty.
     if ((int) request.getDuration() > WHOLE_DAY.duration()) {
-      return query;
+      return Collections.emptyList();
     }
 
-    // Check if there are any events happening at all if so return whole day query.
-    if (events.size() == 0) {
-      query.add(TimeRange.WHOLE_DAY);
-      return query;
+    // Collection of string representing who is to attend the requested meeting.
+    Collection<String> mandatoryAttendees = new ArrayList<>(request.getAttendees());
+
+    // Collection of strings representing who are optional to the meeting.
+    Collection<String> optionalAttendees = new ArrayList<>(request.getOptionalAttendees());
+
+    // If there are no mandatory attendees , return lists of the optional attendees time ranges.
+    if (mandatoryAttendees.isEmpty()) {
+      optionalTimeRanges = populateTimeRanges(events, optionalAttendees, desiredDuration);
+      return optionalTimeRanges;    }
```

Comment:
closing bracket should be on the next line

Reply:
Ack